magicctapipe/scripts/lst1_magic/coincident_brokenGPS.py

```diff
@@ -0,0 +1,418 @@
+                    outfile2.replace("MAGIC", M1_M2),
+                    np.array(
+                        [t_magic_all, time_offset_best, time_offset_best, n_coincident]
+                    ),
+                )
+
+# Create the coincident dl1 files
+magic_dir_name = file_dl1_dir + "/MAGIC/"
+lst_dir_name = file_dl1_dir + "/LST1/"
+
+magic_t = []
+lst_t = []
+start_l, stop_l = [], []
+run_l = []
+obs_id_l = []
+start_m, stop_m = [], []
+run_m = []
+obs_id_m = []
+
+magic_dataset = glob.glob(magic_dir_name + "/*h5")
+magic_dataset = sorted(magic_dataset)
+
+for magic_data in magic_dataset:
+    data_magic = pd.read_hdf(magic_data, key="events/parameters")
+    data_magic["trigger_time"] = (
+        data_magic["time_sec"] + data_magic["time_nanosec"] * 1e-9
+    )
+    start_m.append(min(data_magic["trigger_time"]))
+    stop_m.append(max(data_magic["trigger_time"]))
+    run_m.append(int(data_magic["obs_id"].mean()))
+
+df_magic = pd.DataFrame({"start": start_m, "stop": stop_m, "run": run_m})
+df_magic = df_magic.groupby("run").agg({"start": "min", "stop": "max"}).reset_index()#drop=True)
+
+print(df_magic)
+
+lst_dataset = glob.glob(lst_dir_name + "/*h5")
+lst_dataset = sorted(lst_dataset)
+
+for lst_data in lst_dataset:
+    data_lst = pd.read_hdf(lst_data, key="/dl1/event/telescope/parameters/LST_LSTCam")
+    start_l.append(min(data_lst["trigger_time"]))
+    stop_l.append(max(data_lst["trigger_time"]))
+    run_l.append(int(data_lst["obs_id"].mean()))
+
+df_lst = pd.DataFrame({"start": start_l, "stop": stop_l, "run": run_l})
+df_lst = df_lst.groupby("run").agg({"start": "min", "stop": "max"}).reset_index()#drop=True)
+
+print(df_lst)
+
+
+def print_gti(df1, df2):
+    """
+    This function search the correponting subrun files between LST and MAGIC.
+
+    Parameters
+    ----------
+    df1, df2 : str
+        Input files which contain the timestamo and run/subrun name of MAGIC and LST.
+
+    Returns
+    -------
+    array
+        The array which have a run/subrun ids corresponding the simultaneous observation for MAGIC and LST.
+    """
+    start2_list, stop2_list = (
+        df2["start"],
+        df2["stop"],
+    )
+    i = 0
+    subrun_1, subrun_2 = [], []
+    for run_ in df1["run"].values:
+        df1_ = df1.query("run==@run_")
+        start1, stop1 = df1_["start"], df1_["stop"]
+        start1, stop1 = float(start1.iloc[0]), float(stop1.iloc[0])
+        j = 0
+        for start2, stop2 in zip(start2_list, stop2_list):
+            obs_time_magic = stop1 - start1
+            obs_time_lst = stop2 - start2
+            if [obs_time_magic > obs_time_lst]:
+                s1, e1, s2, e2 = start1, stop1, start2, stop2
+            else:
+                s1, e1, s2, e2 = start2, stop2, start1, stop1
+            if (s1 < s2) & (s2 < e1) == True:
+                subrun_1.append(df1_["run"].values[0])
+                subrun_2.append(df2["run"].values[j])
+            elif (s1 < e2) & (e2 < e1) == True:
```

Comment:
the two cases seem to have the same commands, you could make a single if with ((..) & (...)) | ((..) & (...)) condition